### ISSUE TITLE
st_distance_sphere geography/geometry mismatch

### ISSUE THREAD

**Issue by therealchuckliu** [tminglei/slick-pg#350]
I am getting the error "ERROR: function st_distance_sphere(geography, geometry) does not exist" because my db column is a geography column and I am trying to call distanceSphere using makePoint in the same way that is done in the tests. Unfortunately, that creates a geometry object and I think the error is saying either both need to be geography or geometry, can anyone think of a simple workaround for this?

**Comment by therealchuckliu** [comment]
Interestingly, distance works

**Comment by tminglei** [comment]
Yes, `slick-pg` didn't have good support for `geography`. It's majorly target to `geometry`.

**Comment by naderghanbari** [comment]
Is this still the case? If yes, is a pull request welcome?

**Comment by jdcohen220** [comment]
+1 to geography support

**Comment by tminglei** [comment]
Can you guys recommend any geography java sdk?

_Currently `vividsolutions jts` or `locationtech jts` only support `geometry`._

**Comment by jdcohen220** [comment]
@tminglei I think there is a gootools module for jts that handles geographic types, I have not personally used it: https://docs.geotools.org/latest/javadocs/org/geotools/geometry/jts/JTS.html

**Comment by tminglei** [comment]
@jdcohen220 seems not a good choice.

**Comment by tminglei** [comment]
I finally self-made the geography types by extending jts geometry types, and added geography support .
Can you guys help review and check it?

**Comment by tminglei** [comment]
It's on a new branch `geography`.
If it looks good, I'll merge it to branch `master`.
